src/screens/ChatScreen.tsx

```diff
@@ -0,0 +1,478 @@
+          mentions.push(id);
+          return `\`<!@${id}>[${name}]\``;
+        }),
+      },
+    ];
+    if (messageSelected) {
+      await patchChatMessage(replacedMentionsMessage);
+      setMessageSelected(null);
+    } else {
+      await postChatMessage(replacedMentionsMessage, mentions);
+    }
+    setIsEditMessage(false);
+  };
+
+  const handleMessageOptions = (message: IMessage) => {
+    setMessageSelected(message);
+    handleOpenBottomSheetMessageOptionsModal();
+  };
+
+  const handleEditItemPress = (message: IMessage) => {
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setIsEditMessage(true);
+    setInputMessage(replaceMessageWithMentions(message.text));
+  };
+
+  const handleDeleteItemPress = (messageId: UUID) => {
+    deleteMessage({
+      itemId,
+      messageId,
+    });
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setMessageSelected(null);
+  };
+
+  const handleOnBackDropPressed = () => {
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setIsEditMessage(false);
+    setMessageSelected(null);
+  };
+
+  const renderSuggestions: FC<any> = ({ keyword, onSuggestionPress }) => {
+    if (keyword == null) {
+      return null;
+    }
+    return (
+      <View style={styles.suggestionContainer}>
+        {chatMembers
+          .filter(({ name }) =>
+            name.toLowerCase().includes(keyword.toLowerCase()),
+          )
+          .map(({ id, name }) => (
+            <Pressable
+              key={id}
+              onPress={() => onSuggestionPress({ id, name })}
+              style={{ padding: 10 }}
+            >
+              <Text>{name}</Text>
+            </Pressable>
+          ))}
+      </View>
+    );
+  };
+
+  const renderComposer = (props: any) => {
+    const { text } = props;
+    return (
+      <View style={styles.composerContainer}>
+        {isEditMessage && (
+          <View
+            {...props}
+            style={{
+              width: 60,
+              alignItems: 'center',
+              justifyContent: 'center',
+            }}
+          >
+            <MaterialIcons
+              name="cancel"
+              size={24}
+              color={PRIMARY_COLOR}
+              onPress={() => handleCancelEditMessage()}
+            />
+          </View>
+        )}
+
+        <View style={styles.inputContainer}>
+          <MentionInput
+            {...props}
+            placeholder={t('Type somenthing....')}
+            style={{ paddingLeft: 10 }}
+            value={text}
+            onChange={(value) => setInputMessage(value)}
+            partTypes={[
+              {
+                trigger: MENTION_CHAT_TRIGGER,
+                renderSuggestions,
+                textStyle: { fontWeight: 'bold', color: PRIMARY_COLOR },
+                getPlainString: (mention) => getPlainString(mention),
+              },
+            ]}
+          />
+        </View>
+        <View
+          {...props}
```

Comment:
I do not really like that you spread the props on all the components like this...